frappe/public/js/frappe/assets.js

```diff
@@ -124,19 +124,20 @@ frappe.assets = {
 		// this is virtual page load, only get the the source
 		// *without* the template
 
-		frappe.call({
-			type: "GET",
-			method: "frappe.client.get_js",
-			args: {
-				items: items,
-			},
-			callback: function (r) {
-				$.each(items, function (i, src) {
-					frappe.assets.add(src, r.message[i]);
+		let itemsProcessed = 0;
+		if (items.length) frappe.dom.freeze();
+
+		items.forEach((item, idx) => {
+			fetch(item)
```

Comment:
Few problems with this:

- IN prod non-bundles won't evict the cache unless we do a hard-reload after an upgrade. 
- In dev, same problem but worse because you're actively changing code and it won't refresh unless you do hard reload. 


Possible solutions:
- Send some kind of signal in boot that cache is to be avoided and hard reload is required. So EVERYONE after upgrade will get `window.location.reload(true)`
- Send no-cache header in development mode, add an exception for bundles cause bundles can be cached (invalidated by unique id)

Reply:
> Send some kind of signal in boot that cache is to be avoided and hard reload is required. So EVERYONE after upgrade will get window.location.reload(true)

@ankush I think this is only compatible with firefox?

I've tried this in one of my projects, I could not get it working very well.

Reply:
Would using `frappe.boot.metadata_version` as a version string achieve the same result?